documentation/security-architecture.md

```diff
@@ -0,0 +1,72 @@
+
+## ReaR Security Features
+
+### Protecting Secrets In Configuration Variables
+
+ReaR protects secrets in configuration variables, so that they are not exposed in the (debug) logs. The `--expose-secrets` option can be used to expose secrets in the logs for debugging purposes. This option is disabled by default.
+
+To facilitate protecting secrets even from verbose tracing of the Bash scripts (`set -x`), it is important to wrap statements that handle secrets like this:
+
+```bash
+# set a variable with a secret
+{ VAR='secret_value' ; } 2>>/dev/$SECRET_OUTPUT_DEV
+
+# use the variable with the secret
+{ COMMAND $SECRET_ARGUMENT ; } 2>>/dev/$SECRET_OUTPUT_DEV
+```
+
+### Protecting Against Code Injections
+
+*NOTE: This is our plan, the implementation will follow with the next release. This section will be updated accordingly.*
+
+ReaR is written in the Bash scripting language, the configuration is actually Bash script code and the ReaR system architecture is designed for easily extending ReaR functionality via dropping shell scripts into the ReaR script directories. Therefore the ReaR code is by nature more open than close, which is - from a security perspective - a trade-off for convenience and extensibility over security.
+
+We differentiate between the following scenarios for potential code injections:
+
+1. Reading data
+   1. from trusted operating system configuration files like `/etc/os-release`
+   2. from untrusted configuration files, for example provided by a backup tool that ReaR integrates with
+2. Reading ReaR scripts
+   1. when ReaR is installed
+   2. when ReaR is running in portable mode (`--portable`) or from source checkout
+3. Reading ReaR configuration files in `/etc/rear` or provided via the `-C` option
+
+To mitigate the risk of code injections, the following checks are implemented:
+
+* Trusted operating system configuration files defined as "shell compatible" are read with `source` as we must also support the different quoting styles that can be used there. This is considered safe, as the operating system configuration files are under the control of the operating system vendor and are not writable by unprivileged users.
+* Untrusted configuration files are never read with `source`. Instead, we use `grep` and other tools to extract the required information. This is considered safe, as the configuration files are not executed as code.
+* When ReaR is installed and not run from source or in portable mode, then we validate that ReaR scripts and configuration files are owned by `root` and are not writable by unprivileged users. This is considered safe, as the ReaR scripts are under the control of the ReaR package and are not writable by unprivileged users.
```

Comment:
I would prefer to leave this line out, as it presupposes the solution that we will use. To be concrete, in https://github.com/rear/rear/issues/3259#issuecomment-2385745545 I proposed another solution that matches better what other systems (dynamic linker, dynamic language interpreters) are using.